src/main/java/sirius/kernel/commons/Emojis.java

```diff
@@ -0,0 +1,4838 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.kernel.commons;
+
+/**
+ * Provides helper methods to work with emojis.
+ */
+public class Emojis {
+
+    protected static final CodePointTrie<Boolean> emojiTrie = CodePointTrie.create();
+
+    static {
```

Comment:
**Question:** Would it make sense to extract the list of emojis into a file or config? Or would this just introduce more complexity?

Reply:
That's a good question… I currently tend against it, as it would require additional facilities, and thus increase complexity. To discuss, these would be the options:

- We could add a parser for the [original file](https://unicode.org/Public/emoji/latest/emoji-test.txt), making upgrades simpler but requiring some specialisation for parsing the code points.
- We could use a file format of our own, requiring an conversion when upgrading the list.
- We can leave it in the code, also requiring conversion during upgrades, but at least saving the extra effort. (I have such a script and will provide it separately.)

Additionally, the loading times on my machine are quick enough, I currently don't see a need for lazy initialisation.

So, putting everything together, I would stick to the proposed solution for now, but i will happily change it if required.